app/src/main/java/com/iic/shopingo/api/APIResult.java

```diff
@@ -12,8 +14,15 @@ public APIResult() {
     errorMessage = null;
   }
 
-  public APIResult(String errorMessage) {
-    success = false;
-    this.errorMessage = errorMessage;
+  public APIResult(String error) {
+    errorMessage = error;
+  }
+
+  public APIResult(ApiSimpleResponse response) {
```

Comment:
Why do you need both APIResult and ApiSimpleResponse?
Also let's try to be consistent so name it ApiResult.